TYPO3.Flow/Classes/TYPO3/Flow/Core/Booting/Scripts.php

```diff
@@ -541,22 +546,25 @@ public static function initializeResources(Bootstrap $bootstrap)
     public static function executeCommand($commandIdentifier, array $settings, $outputResults = true, array $commandArguments = array())
     {
         $command = self::buildSubprocessCommand($commandIdentifier, $settings, $commandArguments);
-        $output = array();
-        // Output errors in response
-        $command .= ' 2>&1';
-        exec($command, $output, $result);
-        if ($result !== 0) {
-            if (count($output) > 0) {
-                $exceptionMessage = implode(PHP_EOL, $output);
-            } else {
-                $exceptionMessage = sprintf('Execution of subprocess failed with exit code %d without any further output. (Please check your PHP error log for possible Fatal errors)', $result);
+        $environmentVariables = self::buildFlowEnvironmentVariables($settings);
+        $process = new Process($command, FLOW_PATH_ROOT, $environmentVariables);
+        try {
+            $process->mustRun();
+            $output = $process->getOutput();
+        } catch (ProcessFailedException $processException) {
+            $exceptionMessage = $process->getOutput() . PHP_EOL . $process->getErrorOutput();
+            if ($exceptionMessage === PHP_EOL) {
```

Comment:
wouldn't a trim and strlen count be better here? in case a command exits with error code but no output? no entirely sure that can happen, but this check seems rather flaky